src/jsdoc_transformer.ts

```diff
@@ -0,0 +1,937 @@
+   */
+  disableAutoQuoting?: boolean;
+}
+
+function addCommentOn(node: ts.Node, tags: jsdoc.Tag[], escapeExtraTags?: Set<string>) {
+  const comment = jsdoc.toSynthesizedComment(tags, escapeExtraTags);
+  const comments = ts.getSyntheticLeadingComments(node) || [];
+  comments.push(comment);
+  ts.setSyntheticLeadingComments(node, comments);
+  return comment;
+}
+
+/** @return true if node has the specified modifier flag set. */
+export function isAmbient(node: ts.Node): boolean {
+  let current: ts.Node|undefined = node;
+  while (current) {
+    if (transformerUtil.hasModifierFlag(current, ts.ModifierFlags.Ambient)) return true;
+    current = current.parent;
+  }
+  return false;
+}
+
+type HasTypeParameters =
+    ts.InterfaceDeclaration|ts.ClassLikeDeclaration|ts.TypeAliasDeclaration|ts.SignatureDeclaration;
+
+/** Adds an \@template clause to docTags if decl has type parameters. */
+export function maybeAddTemplateClause(docTags: jsdoc.Tag[], decl: HasTypeParameters) {
+  if (!decl.typeParameters) return;
+  // Closure does not support template constraints (T extends X), these are ignored below.
+  docTags.push({
+    tagName: 'template',
+    text: decl.typeParameters.map(tp => transformerUtil.getIdentifierText(tp.name)).join(', ')
+  });
+}
+
+/** Adds heritage clauses (\@extends, \@implements) to the given docTags for decl. */
+export function maybeAddHeritageClauses(
+    docTags: jsdoc.Tag[], mtt: ModuleTypeTranslator,
+    decl: ts.ClassLikeDeclaration|ts.InterfaceDeclaration) {
+  if (!decl.heritageClauses) return;
+  const isClass = decl.kind === ts.SyntaxKind.ClassDeclaration;
+  const classHasSuperClass =
+      isClass && decl.heritageClauses.some(hc => hc.token === ts.SyntaxKind.ExtendsKeyword);
+  for (const heritage of decl.heritageClauses!) {
+    if (!heritage.types) continue;
+    if (isClass && heritage.token !== ts.SyntaxKind.ImplementsKeyword && !isAmbient(decl)) {
+      // If a class has "extends Foo", that is preserved in the ES6 output
+      // and we don't need to do anything.  But if it has "implements Foo",
+      // that is a TS-specific thing and we need to translate it to the
+      // the Closure "@implements {Foo}".
+      // However for ambient declarations, we only emit externs, and in those we do need to
+      // add "@extends {Foo}" as they use ES5 syntax.
+      continue;
+    }
+    for (const impl of heritage.types) {
+      let tagName = decl.kind === ts.SyntaxKind.InterfaceDeclaration ? 'extends' : 'implements';
+
+      const sym = mtt.typeChecker.getSymbolAtLocation(impl.expression);
+      if (!sym) {
+        // It's possible for a class declaration to extend an expression that
+        // does not have have a symbol, for example when a mixin function is
+        // used to build a base class, as in `declare MyClass extends
+        // MyMixin(MyBaseClass)`.
+        //
+        // Handling this correctly is tricky. Closure throws on this
```

Comment:
I was just poking in this in kythe too: https://github.com/google/kythe/commit/8881f354095d52fd900c503825b0b84d2a7276e8